rules-reviewed/camel3/camel2/xml-dsl-changes.windup.xml

```diff
@@ -0,0 +1,260 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="xml-dsl-changes"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            Rules for changes in XML file (e.g. pom.xml) to run on Apache Camel 3
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="camel" versionRange="[2,3)"/>
+        <targetTechnology id="camel" versionRange="[3,)"/>
+    </metadata>
+    <rules>
+        <rule id="xml-dsl-changes-00001">
+            <when>
+                <or>
+                    <xmlfile matches="//*[(count(b:zip)+count(b:gzip)) >0]">
+                        <namespace prefix="b" uri="http://camel.apache.org/schema/blueprint"/>
+                    </xmlfile>
+                    <xmlfile matches="//*[(count(c:gzip)+count(c:zip)) > 0]">
+                        <namespace prefix="c" uri="http://camel.apache.org/schema/spring"/>
+                    </xmlfile>
+                </or>
+            </when>
+            <perform>
+                <hint title="`zip` and `gzip` dataformat was renamed" effort="1" category-id="mandatory">
+                    <message>`zip` and `gzip` dataformats were renamed to `zipdeflater` and `gzipdeflater`
+
+                    </message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_zip_and_gzip_dataformat"
+                          title="Camel 3 - Migration Guide: Zip and Gzip dafaformats"/>
+                    <quickfix type="REPLACE" name="camel.zipReplacement">
+                        <replacement>zipdeflater</replacement>
+                        <search>zip</search>
```

Comment:
This search will create weird results with the next one. As if there is a `gzip` text in the file it will be replaced to `gzipdeflater` and the next quickfix will replace it as `gzipdeflaterdeflater`